src/Qir/Tests/FullstateSimulator/FullstateSimulatorTests.cpp

```diff
@@ -196,6 +202,9 @@ TEST_CASE("Fullstate simulator: toffoli", "[fullstate_simulator]")
     iqa->ControlledX(2, qs, qs[2]);
     REQUIRE(Result_One == sim->GetResultValue(MZ(iqa, qs[2])));
 
+    iqa->X(qs[1]);
+    iqa->X(qs[0]);
+
```

Comment:
(I'm likely mistaken but to stay on the same page with you guys, just in case, I'll tell about how I interpret this test. Please point for me where exactly I'm wrong)
The line numbers below are from the right column.

After 195 the state is |000> (`qs[0]` is on the left).
.. 197: |100>.
198: |100>.
199: |100>.

201: |110>.
202: |111>.
203: |111>.

205: |101>.
206: |001>.

The subsequent `sim->ReleaseQubit()` should fail for qubit qs[2].

Reply:
You have the right register state, but note also that the last operation on `qs[2]` was a measurement, meaning it is allowed to be released because it is in a well-known state.

Reply:
Thank you for explaining.